lines.go

```diff
@@ -27,14 +27,24 @@ func newLine(line *bill.Line) *Line {
 		Measure:       string(line.Item.Unit.UNECE()),
 		NetUnitPrice:  line.Item.Price.String(),
 		Quantity:      line.Quantity.String(),
-		UnitDiscount:  line.Total.Subtract(line.Sum).Divide(line.Quantity).String(), // not sure if there should be some rescale or matchPrecision
+		UnitDiscount:  unitDiscount(line),
 		NetPriceTotal: line.Total.String(),
 		TaxRate:       line.Taxes[0].Percent.Rescale(2).StringWithoutSymbol(),
 	}
 
 	return Line
 }
 
+func unitDiscount(line *bill.Line) string {
+	if line.Total == line.Sum {
+		return ""
+	}
+
+	discount := line.Total.Subtract(line.Sum).Divide(line.Quantity) // not sure if there should be some rescale or matchPrecision
```

Comment:
Can you not take the Discount amount here and divide by the quantity instead of using the total? Under normal circumstances these types of calculation can easily cause rounding errors, I'd assume however that KSeF are not recalculating the totals.

Reply:
I can't find one Discount amount and calculating the Discount from the array adds more calculations. I removed this part for the time being as this is a optional field. It can be tackled in a separate PR.

Reply:
This is true, but semantically it is more accurate. I'm not worried about adding up, the division is what dramatically increases the chances of rounding errors. I'd imagine the field is optional because most lines don't have discounts, but if a discount is applied, not defining this field might cause problems for the totals. Is there another discount field that can be used?

Reply:
I checked the fields and this is the only field that has to do with discounts. I added a new implementation for the field that is calculated from the discounts.